libgraph/include/katana/PropertyGraph.h

```diff
@@ -114,7 +114,21 @@ class KATANA_EXPORT PropertyGraph {
 
   friend class PGViewCache;
 
-  friend class PropertyGraphRetractor;
+protected:
+  RDG& rdg() { return rdg_; }
+  const RDG& rdg() const { return rdg_; }
+  EntityTypeManager& mutable_node_entity_type_manager() {
+    return node_entity_type_manager_;
+  }
+  EntityTypeManager& mutable_edge_entity_type_manager() {
+    return edge_entity_type_manager_;
+  }
+  EntityTypeIDArray& mutable_node_entity_type_ids() {
```

Comment:
FYI: I've been against exposing NUMAArrays by reference because there's a potential issue of who owns and frees the (memory for) the array. The alternative to this, which I prefer, is providing buffer like objects so that the contents of the array can be changed by calling code, but calling code can't mess with the memory or lifetime of the array.

Reply:
Yeah I wish I didn't have to expose these members, but the manipulation of state by the partitioner requires it and changing partitioner logic is outside the scope of this PR. Will add a TODO though.